packages/@react-aria/select/src/useSelect.ts

```diff
@@ -104,12 +112,30 @@ export function useSelect<T>(props: AriaSelectOptions<T>, state: SelectState<T>,
         triggerProps['aria-label'] && !triggerProps['aria-labelledby'] ? triggerProps.id : null,
         valueId
       ].filter(Boolean).join(' '),
-      onFocus() {
+      onFocus(e: FocusEvent) {
+        if (state.isFocused) {
+          return;
+        }
+
+        if (props.onFocus) {
+          props.onFocus(e);
+        }
+
         state.setFocused(true);
       },
-      onBlur() {
+      onBlur(e: FocusEvent) {
+        if (popoverRef && (e.relatedTarget === ref.current || popoverRef.current?.contains(e.relatedTarget as HTMLElement))) {
```

Comment:
Hey! so after a few discussions, we think this can just be a check for
```suggestion
        if (state.isOpen) {
```

and hopefully we can do this over in combobox as well

Reply:
yeah that works